src/battle_main.c

```diff
@@ -3609,12 +3609,7 @@ static void CheckFocusPunch_ClearVarsBeforeTurnStarts(void)
     }
     TryClearRageStatuses();
     gCurrentTurnActionNumber = 0;
-    {
-        // something stupid needed to match
-        u8 zero;
-
-        gCurrentActionFuncId = gActionsByTurnOrder[(zero = 0)];
-    }
+    gCurrentActionFuncId = gActionsByTurnOrder[gCurrentTurnActionNumber]; // One could and should combine this statement and the one above, but that does not match.
```

Comment:
this comment is worded really poorly

either you meant `gCurrentActionFuncId = gActionsByTurnOrder[gCurrentTurnActionNumber = 0];`, in which case, it's not absolute that one should do this (personally i find it less readable)

or you meant `gCurrentActionFuncId = gActionsByTurnOrder[0];`, in which case 1. just say it should be that and use 0 directly and 2. it's completely insignificant, the pokeemerald one doesn't have a similar comment

Reply:
I resolved that. Thank you!